### PR TITLE
Adding option to print audit results to Slack channel

### PR DESCRIPTION
**Pull request by emmasax1**
Description and Impact
----------------------
This will give the option to print the discrepancies from the audit command into a given slack channel instead of to the output terminal.

Deploy Plan
-----------
> Does Platform Operations need to know anything special about this deploy? Are migrations present?

Rollback Plan
-------------
* To roll back this change, revert the merge with: `git revert -m 1 MERGE_SHA` and perform another deploy.

URLs
----
> Links to bug tickets or user stories.

QA Plan
-------
- [ ] Set up a config file named `~/.aws_auditor.yml`; it should look something like this:
```
slack:
  username: [cool username]
  icon_url: [cool image]
  channel: "#[channel name]"
  webhook: [webhook url]
```
- [ ] Run `bin/aws-auditor audit --slack [profile]` to see it print to the slack channel
- [ ] Run `bin/aws-auditor audit [profile]` to see it print to the terminal as normal, and check that discrepancies match up (slack won't print out anything that is green or blue in terminal)
- [ ] Run `bin/aws-auditor audit --slack [profile]` when MFA is set up and when MFA is not set up
- [ ] Run `bin/aws-auditor audit -si [profile]` to see just the instances print to slack, then again with `bin/aws-auditor audit -sr [profile]` to see just the reserved instances print to slack

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -6,8 +6,6 @@ branches:
 
 language: ruby
 rvm:
-  - 2.0.0
```
Comment (bbergstrom) [inline .travis.yml]: Why the drop of support for Ruby 2.0 and 2.1?  Are you starting to use a 2.2+ only feature?

Comment (emmasax1) [inline .travis.yml]: The config material doesn't work correctly with Ruby versions lower than 2.2. This is known issue, and @matthewrkrieger was planning on trying to figure out why that is.

File: lib/aws_auditor/scripts/audit.rb [hunk 2]
```diff
@@ -19,32 +20,56 @@ def self.execute(environment, options=nil)
...
-        output("EC2Instance", tag_name) if options[:ec2] || no_selection
-        output("RDSInstance", tag_name) if options[:rds] || no_selection
-        output("CacheInstance", tag_name) if options[:cache] || no_selection
+        output("EC2Instance", tag_name, slack, environment) if options[:ec2] || no_selection
+        output("RDSInstance", tag_name, slack, environment) if options[:rds] || no_selection
+        output("CacheInstance", tag_name, slack, environment) if options[:cache] || no_selection
       end
 
-      def self.output(class_type, tag_name)
+      def self.output(class_type, tag_name, slack, environment)
```
Comment (bbergstrom) [inline lib/aws_auditor/scripts/audit.rb]: This method is getting a bit messy and doing too much.  I think we should refactor it a bit to separate the gathering of data and the presentation.  With those two concerns separated, the terminal vs slack output should be easy and succinct.

**Comment by bbergstrom** [comment]
👍

**Pull request merged**